src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -1274,6 +1274,61 @@ private bool SkipExpressionBreakpointsSequences(MethodDefinition methodDefinitio
             return false;
         }
 
+        public bool SkipInlineAssignedAutoProperty(bool skipAutoProps, MethodDefinition methodDefinition, Instruction instruction)
+        {
+            if (!skipAutoProps || !methodDefinition.IsConstructor) return false;
+
+            return SkipGeneratedBackingFieldAssignment(methodDefinition, instruction) ||
+                   SkipDefaultInitializationSystemObject(instruction);
+        }
+
+        private static bool SkipGeneratedBackingFieldAssignment(MethodDefinition methodDefinition, Instruction instruction)
+        {
+            /*
+                For inline initialization of properties the compiler generates a field that is set in the constructor of the class. 
+                To skip this we search for compiler generated fields that are set in the constructor.
+
+                .field private string '<SurName>k__BackingField'
+                .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+                    01 00 00 00
+                )
+
+                .method public hidebysig specialname rtspecialname 
+                instance void .ctor () cil managed 
+                {
+                    IL_0000: ldarg.0
+                    IL_0001: ldsfld    string[System.Runtime] System.String::Empty
+                    IL_0006: stfld     string TestRepro.ClassWithPropertyInit::'<SurName>k__BackingField'
+                    ...
+                }
+                ...
+            */
+            var autogeneratedBackingFields = methodDefinition.DeclaringType.Fields.Where(x =>
+                x.CustomAttributes.Any(ca => ca.AttributeType.FullName.Equals(typeof(CompilerGeneratedAttribute).FullName)) &&
+                x.FullName.Contains("k__BackingField"));
+
+            return instruction.OpCode == OpCodes.Ldarg &&
+                   instruction.Next?.Next?.OpCode == OpCodes.Stfld &&
+                   instruction.Next?.Next?.Operand is FieldReference fr &&
+                   autogeneratedBackingFields.Select(x => x.FullName).Contains(fr.FullName);
+        }
+
+        private static bool SkipDefaultInitializationSystemObject(Instruction instruction)
+        {
+        /*
+            A type always has a constructor with a default instantiation of System.Object. For record types these
```

Comment:
If in future they'll provide record `struct` based we would revisit this. For now I think it's ok.

Reply:
Yes then we have to check again if there are strange sequence points in the constructor.